main/main.cpp

```diff
@@ -2423,6 +2385,22 @@ Error Main::setup(const char *execpath, int argc, char *argv[], bool p_second_ph
 	// always convert to lower case for consistency in the code
 	rendering_driver = rendering_driver.to_lower();
 
+#ifdef METAL_ENABLED
+	if (rendering_driver == "auto") {
+		rendering_driver = "metal";
+	}
+#endif
+#ifdef VULKAN_ENABLED
+	if (rendering_driver == "auto") {
+		rendering_driver = "vulkan";
+	}
+#endif
+#ifdef D3D12_ENABLED
+	if (rendering_driver == "auto") {
+		rendering_driver = "d3d12";
+	}
```

Comment:
This would change behavior for Windows and make it default to d3d12.

Reply:
Vulkan check is first, so it will only change behavior if it's built without Vulkan support.